test/src/unit/modules/ProjectList.test.js

```diff
@@ -0,0 +1,297 @@
+const ProjectList = require('../../../../src/pfe/portal/modules/ProjectList');
+const ProjectListError = require('../../../../src/pfe/portal/modules/utils/errors/ProjectListError');
+
+const assert = require('assert');
+const chai = require('chai');
+const chaiSubset = require('chai-subset');
+const chaiAsPromised = require('chai-as-promised');
+chai.use(chaiSubset);
+chai.use(chaiAsPromised);
+chai.should();
+
+
+describe('ProjectList.js', () => {
+    before(() => {
+        global.codewind = { CODEWIND_WORKSPACE: `${__dirname}/projectlist_temp` };
+        fs.ensureDirSync(global.codewind.CODEWIND_WORKSPACE);
+    });
+    after(() => {
+        fs.removeSync(global.codewind.CODEWIND_WORKSPACE);
+    });
+    describe('new ProjectList', () => {
+        it('Initialises a new, empty ProjectList', () => {
+            const projectList = new ProjectList();
+            projectList.should.be.an('object');
+            projectList.list.should.deep.equal({});
+        });
+    });
+    describe('addProject(project)', () => {
+        it('Creates a new Project and adds it to the ProjectList', () => {
+            const projectList = new ProjectList();
+            const project = new Project({ name: 'newdummyproject' }, global.codewind.CODEWIND_WORKSPACE);
+            projectList.addProject(project);
+            const list = projectList.list;
+            list.should.not.eql({});
+            Object.keys(list).length.should.equal(1);
+            list.should.have.property(project.projectID);
+            
+        });
+        it('Throws an error as the project already exists', () => {
+            const projectList = new ProjectList();
+            const project = new Project({ name: 'newdummyproject' }, global.codewind.CODEWIND_WORKSPACE);
+            projectList.addProject(project);
+            const list = projectList.list;
+            list.should.not.eql({});
+            Object.keys(list).length.should.equal(1);
+            list.should.have.property(project.projectID);
+
+            const error = new ProjectListError('ALREADY_EXISTS', project.projectID);
+            assert.throws(() => projectList.addProject(project), error);
+        });
+    });
+    describe('projectNameExists(name)', () => {
+        it('Returns true as project name exists', () => {
+            const projectList = new ProjectList();
+            const project = new Project({ name: 'newdummyproject' }, global.codewind.CODEWIND_WORKSPACE);
+            projectList.addProject(project);
+            const exists = projectList.projectNameExists('newdummyproject');
+            exists.should.be.true;
+        });
+        it('Returns false as ProjectList is empty', () => {
+            const projectList = new ProjectList();
+            const exists = projectList.projectNameExists('random');
+            exists.should.be.false;
+        });
+        it('Returns false as project name does not exist', () => {
```

Comment:
For a few of these I'd favour `returns false when project does not exist' etc... i.e when over as. as makes it sound a bit like the tests rely on one another